lib/react_client.dart

```diff
@@ -36,19 +36,92 @@ newJsMap(Map map) {
 }
 
 /**
- * Type of [children] must be child or list of childs, when child is JsObject or String
+ * Type of [children] must be child or list of children, when child is JsObject or String
  */
 typedef JsObject ReactComponentFactory(Map props, [dynamic children]);
 typedef Component ComponentFactory();
 
 class ReactComponentFactoryProxy implements Function {
-  final ReactComponentFactory _call;
   final JsFunction reactComponentFactory;
-  ReactComponentFactoryProxy(this.reactComponentFactory, this._call);
+  ReactComponentFactoryProxy(this.reactComponentFactory);
 
   JsObject call(Map props, [dynamic children]) {
-    return this._call(props, children);
+    List<dynamic> reactParams = [_generateExtendedJSProps(props, children)];
+    reactParams.add(children);
+    return reactComponentFactory.apply(reactParams);
   }
+
+  dynamic noSuchMethod(Invocation invocation) {
+    if ((invocation.memberName == #call) && (invocation.isMethod)) {
+      Map props = invocation.positionalArguments[0];
+      List<dynamic> children = invocation.positionalArguments.sublist(1);
+      List<dynamic> reactParams = [_generateExtendedJSProps(props, children)];
+      reactParams.addAll(children);
+      return reactComponentFactory.apply(reactParams);
+    }
+    return super.noSuchMethod(invocation);
+  }
+
+  JsObject _generateExtendedJSProps(Map props, dynamic children) {
+
+    Map extendedProps = new Map.from(props);
+    extendedProps['children'] = children;
```

Comment:
Is everyone okay with that?

I also think that this kind of change should be accompanied by the addition of `React.children` utilities in Dart.

I could go either way on with changing `children` to be an opaque data structure like it is in JS, but perhaps we should continue converting it to a `List`, and address that in a separate PR (which may or may not be performance related).